auth/standard_role_manager.cc

```diff
@@ -248,27 +248,26 @@ future<> standard_role_manager::start() {
             }
             return make_ready_future<>();
         }).then([this] {
-            _stopped = auth::do_after_system_ready(_as, [this] {
-                return seastar::async([this] {
-                    if (legacy_mode(_qp)) {
-                        _migration_manager.wait_for_schema_agreement(_qp.db().real_database(), db::timeout_clock::time_point::max(), &_as).get();
-
-                        if (any_nondefault_role_row_satisfies(_qp, &has_can_login).get()) {
-                            if (legacy_metadata_exists()) {
-                                log.warn("Ignoring legacy user metadata since nondefault roles already exist.");
-                            }
-
-                            return;
-                        }
+            _stopped = auth::do_after_system_ready(_as, [this] () mutable -> future<void> {
+                if (legacy_mode(_qp)) {
+                    co_await _migration_manager.wait_for_schema_agreement(_qp.db().real_database(), db::timeout_clock::time_point::max(), &_as);
 
+                    if (co_await any_nondefault_role_row_satisfies(_qp, &has_can_login)) {
                         if (legacy_metadata_exists()) {
-                            migrate_legacy_metadata().get();
-                            return;
+                            log.warn("Ignoring legacy user metadata since nondefault roles already exist.");
                         }
+
+                        co_return;
                     }
-                    create_default_role_if_missing().get();
-                });
+
+                    if (legacy_metadata_exists()) {
+                        co_await migrate_legacy_metadata();
+                        co_return;
```

Comment:
hint, since co_await returns future you can do in one line, co_return co_await ...

Reply:
IMO it's clearer this way. co_return co_await tells the reader we're calling migrate_legacy_metadata() and returning whatever it returns. Then the reader sees it returns void and has to feel stupid.

Returning void is useful for templates (which don't know what they return) but is bad for readability.

Reply:
at least it wouldn't compile if migrate_legacy_metadata returns something, current way best case there will be warning

but not important change anyway